src/server/worker/pipeline/transform/transform.proto

```diff
@@ -13,53 +13,48 @@ message DatumSet {
   // Inputs
   string job_id = 1 [(gogoproto.customname) = "JobID"];
   string file_set_id = 2;
-  pfs_v2.Commit output_commit = 3;
+  pfs_v2.PathRange path_range = 3;
+  pfs_v2.Commit output_commit = 4;
 
   // Outputs
-  string output_file_set_id = 4;
-  string meta_file_set_id = 5;
-  datum.Stats stats = 6;
+  string output_file_set_id = 5;
+  string meta_file_set_id = 6;
+  datum.Stats stats = 7;
 }
 
-message UploadDatumsTask {
-  pps_v2.Job job = 1;
-} 
-
-message UploadDatumsTaskResult {
-  string file_set_id = 1;
-  int64 count = 2;
-} 
-
 message ComputeParallelDatumsTask {
-  pps_v2.Job job = 1;
+  pps_v2.JobInfo job_info = 1;
   string file_set_id = 2;
   string base_file_set_id = 3;
+  pfs_v2.PathRange path_range = 4;
 } 
 
 message ComputeParallelDatumsTaskResult {
   string file_set_id = 1;
+  datum.Stats stats = 2;
 }
 
 message ComputeSerialDatumsTask {
-  pps_v2.Job job = 1;
+  pps_v2.JobInfo job_info = 1;
   string file_set_id = 2;
   pfs_v2.Commit base_meta_commit = 3;
   bool no_skip = 4;
+  pfs_v2.PathRange path_range = 5;
 }
 
 message ComputeSerialDatumsTaskResult {
   string file_set_id = 1;
-  string delete_file_set_id = 2;
-  int64 skipped = 3;
+  string output_delete_file_set_id = 2;
+  string meta_delete_file_set_id = 3;
+  datum.Stats stats = 4;
 } 
 
 message CreateDatumSetsTask {
-  pps_v2.Job job = 1;
-  pfs_v2.Commit output_commit = 2;
-  string file_set_id = 3;
+  string file_set_id = 1;
+  pfs_v2.PathRange path_range = 2;
```

Comment:
I think this can be solved with more documentation, but I think that should be in the followup that removes parallel jobs. I am planning on shuffling the registry / pending job code a bit more.